src/glium_renderer.rs

```diff
@@ -67,21 +68,33 @@ impl Renderer {
             device_objects: device_objects,
         })
     }
+
     pub fn render<'a, S: Surface>(&mut self, surface: &mut S, ui: Ui<'a>) -> RendererResult<()> {
         let _ = self.ctx.insert_debug_marker("imgui-rs: starting rendering");
-        let result = ui.render(|draw_list| self.render_draw_list(surface, draw_list));
+        let result = ui.render(|ui, draw_list| self.render_draw_list(surface, ui, draw_list));
         let _ = self.ctx.insert_debug_marker("imgui-rs: rendering finished");
         result
     }
+
     fn render_draw_list<'a, S: Surface>(&mut self,
                                         surface: &mut S,
+                                        ui: &'a Ui<'a>,
                                         draw_list: DrawList<'a>)
                                         -> RendererResult<()> {
+        use glium::{Blend, DrawParameters, Rect};
+        use glium::uniforms::MagnifySamplerFilter;
+
         try!(self.device_objects.upload_vertex_buffer(&self.ctx, draw_list.vtx_buffer));
         try!(self.device_objects.upload_index_buffer(&self.ctx, draw_list.idx_buffer));
 
-        let (width, height) = surface.get_dimensions();
-        let matrix = [[2.0 / (width as f32), 0.0, 0.0, 0.0],
+        let (width, height) = ui.imgui().display_size();
+        let (scale_width, scale_height) = ui.imgui().display_framebuffer_scale();
+
+        if width == 0.0 || height == 0.0 {
+            return Ok(());
+        }
+
+        let matrix = [[2.0 / width as f32, 0.0, 0.0, 0.0],
                       [0.0, 2.0 / -(height as f32), 0.0, 0.0],
                       [0.0, 0.0, -1.0, 0.0],
                       [-1.0, 1.0, 0.0, 1.0]];
```

Comment:
Tried to keep this in line with: https://github.com/ocornut/imgui/commit/e58f99179aa3e8fa78e783f772c31ee23d40863d#diff-47471da84bd4194a17a350e728b638b1R45